distributed/tests/test_system_monitor.py

```diff
@@ -113,3 +113,17 @@ def test_gil_contention():
         sm = SystemMonitor()
         a = sm.update()
         assert "gil_contention" in a
+
+    sm.close()
+    sm.close()  # Idempotent
+
+    # Closing SystemMonitor will stop gilknocker thread
+    class FakeGilKnocker:
```

Comment:
Do we need a patch like this, or is there some state/attribute we can check on the actual gil monitor to confirm it's been stopped?

Reply:
Nice point, added in the update to gilknocker, and updated to check that. `is_running` attr